### DISCUSSION TITLE
Some way to change a param when other params change

### DISCUSSION THREAD

**Discussion by ak4zh** [paoloricciuti/sveltekit-search-params#136] [General]
I am using `sveltekit-search-params@next` to sync my form inputs to URL search params.
I am looking for a way to send user back to page 1 if he changes any search params but not the page param itself.
I already tried to use $effect and beforeNavigate with no success.

Here's a quick setup: https://www.sveltelab.dev/80c1fhlh6bnr8nb

**Comment by paoloricciuti** [comment]
Mmm the effect one seems like a bug but i have to check...i've tried exploring it a bit and it seems like the value of the url returned from the page store doesn't seem to reflect the actual url.

**Reply by ak4zh** [reply]
I have added another variation of $effect using untrack and untracking `params.page`.

This works only when the dependency params are not null / non empty (?page=1&brand=audi&model=a1&q=abc)
When the dependency params are null it stops working, technically it should not trigger effect if you have no params and just doing page navigate. (?page=1, ?page=2), but it does run the effect.

```js
$effect(() => {
		console.log(params.brand, params.model, params.q)
		untrack(() => {
			params.page = '1'
		})
	})
	
```

**Reply by paoloricciuti** [reply]
Uh this might be me being silly billy with loose equality. Will check
